src/JoiForm.js

```diff
@@ -115,19 +101,14 @@ class JoiForm extends Component {
   }
 
   componentDidMount() {
-    let  schema = {};
-    if (this.props.schema) {
-      schema = reduce(this.props.schema, (acc, x) => ({...acc, [x._meta.name || camelize(x._flags.label)]: x}), {})
-    }
+    const  schema = reduce(this.props.schema, (acc, x) => ({...acc, [x._meta.name || camelize(x._flags.label)]: x}), {})
```

Comment:
Remove componentDidMount and move that functionality in the constructor to remove a useless rerendering